jcapi-tags.go

```diff
@@ -127,15 +111,16 @@ func (jc JCAPI) GetAllTags() (tagList []JCTag, err JCError) {
 }
 
 func (jc JCAPI) GetTagByName(tagName string) (tag JCTag, err JCError) {
-	url := fmt.Sprintf("/tags/%s", tagName)
+	url := fmt.Sprintf("%s/%s", TAGS_PATH, tagName)
 
-	retVal, err := jc.Get(url)
+	tags, err := jc.GetTagsByUrl(url)
 	if err != nil {
-		err = fmt.Errorf("ERROR: Could not get tag by name/id '%s', err='%s'", tagName, err)
+		err = fmt.Errorf("ERROR: Could not get tags by name for '%s', url='%s', err='%s'", tagName, url, err.Error())
+		return
 	}
 
-	if retVal != nil {
-		jc.getTagFieldsFromInterface(retVal.(map[string]interface{}), &tag)
+	if len(tags) > 0 {
+		tag = tags[0]
 	}
 
 	return
```

Comment:
What is the desired behavior if one calls `GetTagByName` with a name that doesn't exist? Right now I think it returns an error that has some context around '404', but if `GetTagsByUrl` could return no err but an empty tags, you could exit from this function with `nil, nil` as well; if we want 404 for missing tag, we should probably instead check `len(tags) == 0` and return an error, otherwise return `tags[0].